src/PPU.cpp

```diff
@@ -163,14 +134,14 @@ void PPU::updateTileData() {
 			uint8_t lo2 = bus->read(block2_start + i * 16 + j * 2);
 			uint8_t hi2 = bus->read(block2_start + i * 16 + j * 2 + 1);
 
-			set_line(block0, x * 8, j + y * 8, hi0, lo0);
-			set_line(block1, x * 8, j + y * 8, hi1, lo1);
-			set_line(block2, x * 8, j + y * 8, hi2, lo1);
+			set_line(x * 8, j + y * 8, hi0, lo0);
+			set_line(x * 8, 64 + j + y * 8, hi1, lo1);
+			set_line(x * 8, 128 + j + y * 8, hi2, lo2);
 		}
 	}
 }
 
-void PPU::updateTileMap() {
+void PPU::updateTileMap(uint32_t* pixelBuffer) {
```

Comment:
I'm passing in the pixel buffer than should be manipulated. This way I can create it in the SDL render function and have access to it in the main loop. I could also maybe have it as a variable in the PPU that I access before locking the texture, pass it into LockTexture, and then call updateTileMap while the texture is locked. This way seems fine for now but I'll reassess if it simplifies porting or if it makes more sense.